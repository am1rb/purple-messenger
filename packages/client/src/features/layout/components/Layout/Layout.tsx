import React, { memo, ReactNode } from 'react';
import AppMenu from 'features/layout/components/AppMenu';
import { ClassesType } from 'core/type/general';
import useStyles from './Layout.styles';

interface Props {
  children: ReactNode;
  classes?: ClassesType<typeof useStyles>;
}

function Layout({ children, classes: classesProp }: Props) {
  const classes = useStyles({ classes: classesProp });
  return (
    <div className={classes.root}>
      <div className={classes.container}>
        <AppMenu className={classes.appMenu} />
        <div className={classes.content}>{children}</div>
      </div>
    </div>
  );
}

export default memo(Layout);
