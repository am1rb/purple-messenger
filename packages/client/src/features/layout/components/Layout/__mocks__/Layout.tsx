import React from "react";

interface LayoutProps {
  children: React.ReactNode;
}

const layout = jest.fn(({ children }: LayoutProps) => (
  <div data-testid="layout">{children}</div>
));

export default layout;
