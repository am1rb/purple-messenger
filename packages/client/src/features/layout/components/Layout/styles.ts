import { makeStyles, Theme } from '@material-ui/core';

export default makeStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    alignContent: 'center',
    height: '100%',
    padding: '20px 0'
  },
  container: {
    maxWidth: 1200,
    margin: '0 auto',
    height: '100%',
    width: '100%',
    backgroundColor: '#e3e3f0',
    borderRadius: theme.shape.borderRadius,
    display: 'flex'
  },
  appMenu: {
    flexGrow: 0,
    flexShrink: 0
  },
  content: {
    flexGrow: 1,
    flexShrink: 1,
    overflow: 'hidden',
  }
}));
