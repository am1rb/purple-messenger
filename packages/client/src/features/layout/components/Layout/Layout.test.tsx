import React from "react";
import { render } from "@testing-library/react";
import Layout from ".";

jest.mock("features/layout/components/AppMenu");

describe("The <Layout /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<Layout>layout children</Layout>);
    expect(container).toMatchSnapshot();
  });
});
