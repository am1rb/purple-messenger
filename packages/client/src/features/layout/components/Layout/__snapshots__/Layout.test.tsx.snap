// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Layout /> tests Should match the snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="container"
    >
      <div
        data-testid="mock-app-menu"
      />
      <div
        class="content"
      >
        layout children
      </div>
    </div>
  </div>
</div>
`;
